backend/app/status_manager.py

```
@@ -0,0 +1,203 @@
+from .utils import normalize_url # Import from utils
+
+logger = logging.getLogger(__name__)
+
+# Initialize a multiprocessing Manager and create a managed dictionary
+# This needs to be at the module level to be accessible by different processes/workers
+try:
+    manager = Manager()
+    crawl_jobs_managed = manager.dict()
+    logger.info("Initialized multiprocessing Manager and managed dictionary for crawl_jobs.")
+except Exception as e:
+    logger.error(f"Failed to initialize multiprocessing Manager: {e}. Falling back to regular dict (STATE WILL NOT BE SHARED BETWEEN PROCESSES).", exc_info=True)
```

Ruff (0.8.2)

Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)